inst/include/dust/filter_tools.hpp

```diff
@@ -59,16 +59,55 @@ real_type scale_log_weights(typename std::vector<real_type>::iterator w, size_t
       max_w = std::max(max_w, *wi);
     }
   }
+
+  if (max_w == -std::numeric_limits<real_type>::infinity()) {
+    return max_w;
+  }
+
   real_type tot = 0.0;
   wi = w;
   for (size_t i = 0; i < n; ++i, ++wi) {
     *wi = std::exp(*wi - max_w);
     tot += *wi;
   }
+
   return std::log(tot / n) + max_w;
 }
 
 
+template <typename real_type>
+bool early_exit(const std::vector<real_type>& log_likelihood,
```

Comment:
I was a bit confused as to why these are both vectors, but I guess that is for `n_pars > 1`? Could we add a comment on that here maybe

Reply:
yeah, this could do with a comment, will add one